src/sys/reboot.rs

```diff
@@ -0,0 +1,50 @@
+//! Reboot/shutdown or enable/disable Ctrl-Alt-Delete.
+
+use {Errno, Error, Result};
+use libc;
+use void::Void;
+use std::mem::drop;
+
+/// How exactly should the system be rebooted.
+///
+/// See [`set_cad_enabled()`](fn.set_cad_enabled.html) for
+/// enabling/disabling Ctrl-Alt-Delete.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum RebootMode {
```

Comment:
(Somehow a comment went missing yesterday; I might have forgotten to press comment).

What problem did occur when you tried to use `#[repr(i32)]`? All constants have type `libc::c_int` which is an alias for `i32`. The enumeration `Signal` in `sys/signal.rs` does the same. You could then get rid of the match bellow.
As you can see there, I used the name of the corresponding constants for the elements of the enumeration. I would prefere if we did that here as well, even though we have no convention yet. However, bitflags types use this convention, so it would be consistent.